llvm/lib/CodeGen/CodeGenPrepare.cpp

```diff
@@ -8396,7 +8396,29 @@ bool CodeGenPrepare::optimizeInst(Instruction *I, ModifyDT &ModifiedDT) {
     return true;
 
   switch (I->getOpcode()) {
-  case Instruction::Shl:
+  case Instruction::Shl: {
+    // shl X, cttz(Y) -> mul (Y & -Y), X if cttz is unsupported on the target.
```

Comment:
`shl X, cttz(Y) -> mul (Y & -Y), X` -> `shl X, cttz(Y) -> X, mul (Y & -Y)`?
I missed the `X` at first galance.

Reply:
I put `X` in RHS because it may be a constant.